src/driver_zigbee.cpp

```diff
@@ -0,0 +1,469 @@
+      char buf[128];
+      std::snprintf(buf, sizeof buf, "network pjoin %d",
+                    zigbee_msg.network_mgmt_cmd().permit_join_params().time());
+      command = buf;
+    } else if (zigbee_msg.network_mgmt_cmd().type() ==
+               ZigBeeMsg::NetworkMgmtCmd::DISCOVERY_INFO) {
+      command = "plugin device-database print-all";
+    } else if (zigbee_msg.network_mgmt_cmd().type() ==
+               ZigBeeMsg::NetworkMgmtCmd::RESET_PROXY) {
+      // Kill the ZigBeeGateway app
+      system("sudo pkill ZigBeeGateway");
+
+      // Starting the app
+      system(
+          "sudo /usr/share/admobilize/matrix-creator/blob/ZigBeeGateway -n 1 "
+          "-p ttyS0 -v &");
+
+      int i = 0;
+      const int count = 10;
+      tcp_client_.reset(new TcpClient());
+
+      std::cerr << "Trying to connect with the Gateway ...." << std::endl;
+      zmq_push_error_->Send("Trying to connect with the Gateway ....");
+
+      for (i = 0; i < count; ++i) {
+        // Sleep 0.5 sec to wait for
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+        if (tcp_client_->Connect(gateway_ip, gateway_port)) {
+          std::cerr << "Connected to the Gateway." << std::endl;
+          zmq_push_error_->Send("Connected to the Gateway.");
+          break;
+        } else {
+          if (i == count - 1) {
+            std::cerr << "No connection to the Gateway" << std::endl
+                      << std::flush;
+            zmq_push_error_->Send("No connection to the Gateway");
+          }
+          std::cerr << "Trying to connect with the Gateway ...." << std::endl;
+          zmq_push_error_->Send("Trying to connect with the Gateway ....");
+        }
+      }
+
+      std::cerr.flush();
+
+    } else if (zigbee_msg.network_mgmt_cmd().type() ==
+               ZigBeeMsg::NetworkMgmtCmd::IS_PROXY_ACTIVE) {
+      command = "";  // No need to send message to Gateway
+      if (tcp_client_->GetErrorMessage().compare("Connected") == 0) {
+        zigbee_msg.mutable_network_mgmt_cmd()->set_is_proxy_active(true);
+      } else {
+        zigbee_msg.mutable_network_mgmt_cmd()->set_is_proxy_active(false);
+      }
+
+      std::string buffer;
+      zigbee_msg.SerializeToString(&buffer);
+      zqm_push_update_->Send(buffer);
+
+    } else if (zigbee_msg.network_mgmt_cmd().type() ==
+               ZigBeeMsg::NetworkMgmtCmd::NETWORK_STATUS) {
+      command = "info";
+    }
+
+    // Sending the messsage
+    if (command != "") {
+      if (tcp_client_.get() != nullptr) {
+        tcp_client_->Send(command + "\n");
+      }
+    }
+  }
+
+  return true;
+}
+
+ZigBeeMsg zigbee_msg;
+bool bulding_discovery_result = false;
+
+bool ZigbeeDriver::SendUpdate() {
+  std::string line;
+  while (tcp_client_->GetLine(&line)) {
+    line = Trim(line);
+    // std::cerr << ">>" << line << std::endl;
+
+    const char network_state_line[] = "network state";
+    std::size_t found = line.find(network_state_line);
```

Comment:
You could save all the assignations to "found" and just do:

```
if (line.find(network_state_line) != std::string::npos)
```

Up to you. OK the way it is. The suggestion is because I had to scan other parts of the file to make sure that "found" wasn't being used for something else.